### ISSUE TITLE
Unable to exec a created singularity image as a non-root user

### ISSUE THREAD

**Issue by jcohen02** [singularityware/docker2singularity#66]
Hi, I've previously been able to generate Singularity images using a much older version of the `docker2singularity.sh` script that I can then subsequently run as a non-root user.

Trying the latest master version, I can build the `docker2singularity.sh` docker container and run the tool to get a Singularity image file but I can only access the generated image as root due to the permissions of most of the files in the `/.singularity.d/env` directory. `10-docker.sh` seems to have permissions set to 755 but all the other scripts have their permissions set to 700 resulting in the following error when trying to access the container as a non-root user:

```
/.singularity.d/actions/exec: .: line 7: can't open '/.singularity.d/env/01-base.sh'
```

I'm testing this using the `alpine:3.7` docker image as follows (as root):

```
$ docker run -v /var/run/docker.sock:/var/run/docker.sock -v /tmp/test:/output --privileged -t --rm docker2singularity -n testimage alpine:3.7
```

The `testimage.simg` image file is created in /tmp/test and I can successfully run it using:
```
$ singularity exec /tmp/test/testimage.simg /bin/sh
```

If I switch back to a non root user, it fails to run giving the error shown above. I am also able to successfully run the docker command to create the image as a non-root user that is a member of the docker group but the permissions of the files in `/.singularity.d/env` still prevent the image from being run while not root.

I'm running singularity version 3.3.0-1.

Apologies if I'm missing something here, or if this is intended behaviour, I couldn't see any information about this in the docs or previous issues. Thanks.

**Comment by vsoch** [comment]
You shouldn't need to run the docker container as root, but you would need the user added to the Docker group. But there is still the issue with the files permissions.

Could you try rebuilding the container, but commenting these two lines that copy actions and environment scripts? https://github.com/singularityware/docker2singularity/blob/master/docker2singularity.sh#L273

Older versions of Singularity wouldn't provide these, but (I think) it's likely the case that the build will handle it. Let me know if the files exist, and if that resolves the permissions, and I can do a fix.

**Comment by jcohen02** [comment]
Thanks for the quick response @vsoch. 

I commented out [lines 273 and 274](https://github.com/singularityware/docker2singularity/blob/a093e80e7591af9fa39f6b6b589b8b830180a2af/docker2singularity.sh#L273-L274) in the `docker2singularity.sh` script. 

I hadn't noticed that the subsequent commands generate the `10-docker.sh` script and copy this into `$build_sandbox/.singularity.d/env/` so the first time I ran, the run fails at step 5 with:

```
(5/10) Setting ENV variables...
cp: can't create '/tmp/testimage.build/.singularity.d/env/10-docker.sh': No such file or directory
```

I added a line to create this directory after the two commented out lines:
```
mkdir $build_sandbox/.singularity.d/env; chmod 755 $build_sandbox/.singularity.d/env
```

The script now runs successfully, creates the testimage.simg file and I can run a terminal in the singularity container without having to be root.

I can also confirm that within the singularity container, `/.singularity.d/env` contains 6 scripts in addition to `10-docker.sh` (i.e. `01-base.sh`, `90-environment.sh`, etc.)

There are also 5 scripts in the `/.singularity.d/actions` directory. It looks like these directories are therefore being populated correctly without the copy commands in script.

If you don't mind updating this, that's great, otherwise I'm happy to raise a PR for the changes.

**Comment by vsoch** [comment]
Woot, good work @jcohen02 I'm so glad that fixed it! 

If you have the bandwidth, I'd greatly appreciate a PR to the branch that corresponds with the version you are working with (e.g., https://github.com/singularityware/docker2singularity/tree/v3.3.0). Once that goes through, I'll open another PR for that branch to master.

**Comment by jcohen02** [comment]
Thanks @vsoch, great that this was straightforward to resolve. PR created against v3.3.0.

**Comment by vsoch** [comment]
Fixed with #68 #69 thanks @jcohen02 !
